### PR TITLE
[CARBONDATA-1439] Wrong Error message shown for Bad records even when BAD_RECORDS_LOGGER_ENABLE is set to true

### PR DESCRIPTION
**Pull request by shivangi1015**
This PR corrects the error message for bad records scenario when BAD_RECORDS_LOGGER_ENABLE is set to true.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by sraghunandan** [comment]
Do not change the format.keep @override in previous line

**Comment by shivangi1015** [comment]
I have corrected this.

File: processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java [hunk 1]
```diff
@@ -158,8 +158,10 @@ public CarbonRow convert(CarbonRow row) throws CarbonDataLoadingException {
       if (!logHolder.isLogged() && logHolder.isBadRecordNotAdded()) {
         badRecordLogger.addBadRecordsToBuilder(copy.getData(), logHolder.getReason());
         if (badRecordLogger.isDataLoadFail()) {
-          String error = "Data load failed due to bad record: " + logHolder.getReason() +
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java]: What is wrong with this message?

**Comment by ravipesala** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by manishgupta88** [comment]
@shivangi1015 ...kindly rebase this PR..it has merge conflicts

File: processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java (hunk 1 above)
Comment (shivangi1015) [reply processing/src/main/java/org/apache/carbondata/processing/newflow/converter/impl/RowConverterImpl.java]: Even when the BAD_RECORDS_ENABLE is set to true, the error message includes this: enable the bad record logger to know the detail reason.

**Comment by sraghunandan** [comment]
retest this please

**Comment by manishgupta88** [comment]
add to whitelist

**Comment by manishgupta88** [comment]
retest this please

**Comment by ravipesala** [comment]
add to whitelist

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2451/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2187/

**Comment by manishgupta88** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/985/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**